src/main/java/org/jenkinsci/plugins/workflow/actions/ErrorAction.java

```diff
@@ -170,28 +182,48 @@ public String getUrlName() {
     @Restricted(Beta.class)
     public static boolean equals(Throwable t1, Throwable t2) {
         if (t1 == t2) {
+            LOGGER.fine(() -> "Same object: " + t1);
             return true;
-        } else if (t1.getClass() != t2.getClass()) {
+        }
+        String id1 = findId(t1, new HashSet<>());
```

Comment:
Going to review changes to this method to see how much is actually necessary now that the `ErrorAction.<init>` change is fixed.

Reply:
Something like these changes still seems necessary or it always fails with

```
Different types: class hudson.remoting.RequestAbortedException vs. class hudson.remoting.ProxyException
```

c27b80bcdfb3d416e66ae24f2e2cac65b12f6222 would also work I guess, saving a pair of `HashSet` constructions.